File: services/content-service/src/main/java/com/blog/content/entity/BlogBot.java
package com.blog.content.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("blog_bot")
public class BlogBot {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String name;
    private String avatar;
    private String style;
    private Long mainTagId;
    private String summaryStyle;
    private String wordCountPreference;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/api/blogBot.ts
import request from './request'

export interface BlogBotItem {
  id: number
  name: string
  avatar?: string | null
  style: string
  mainTagId?: number | null
  mainTagName?: string | null
  summaryStyle: string
  wordCountPreference: string
  createdAt: string
  updatedAt: string
}

export interface CreateBlogBotBody {
  name: string
  avatar?: string
  style: string
  mainTagId?: number
  summaryStyle: string
  wordCountPreference: string
}

/**
 * 获取当前用户的博客机器人列表
 */
export function getBlogBotsMe(): Promise<BlogBotItem[]> {
  return request.get('blog-bots/me').then((data) => (data as unknown) as BlogBotItem[])
}

/**
 * 创建博客机器人
 */
export function createBlogBot(body: CreateBlogBotBody): Promise<BlogBotItem> {
  return request.post('blog-bots', body).then((data) => (data as unknown) as BlogBotItem)
}
